homeassistant/components/lgthinq/switch.py

```diff
@@ -0,0 +1,197 @@
+    return POWER_ON if bool(value) else POWER_OFF
+
+
+@dataclass(kw_only=True, frozen=True)
+class ThinQSwitchEntityDescription(ThinQEntityDescription, SwitchEntityDescription):
+    """The entity description for switch."""
+
+
+OPERATION_SWITCH_DESC: dict[ThinQProperty, ThinQSwitchEntityDescription] = {
+    ThinQProperty.AIR_FAN_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.AIR_FAN_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.AIR_FAN_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.AIR_PURIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.AIR_PURIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.AIR_PURIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.BOILER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.BOILER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.BOILER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.DEHUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.DEHUMIDIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.DEHUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.HUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+}
+
+AIR_PURIFIER_FAN_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.AIR_FAN_OPERATION_MODE],
+)
+AIRPURIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.AIR_PURIFIER_OPERATION_MODE],
+)
+DEHUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.DEHUMIDIFIER_OPERATION_MODE],
+)
+HUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.HUMIDIFIER_OPERATION_MODE],
+)
+SYSTEM_BOILER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.BOILER_OPERATION_MODE],
+)
```

Comment:
Will there be more switches for these devices in the future?

It feels a bit repeating to first have a dict that can transform property into entity description, and then a list for switch per device type and then a way to get from device type to list?

Reply:
Refactored